src/library/components/table/table-row.tsx

```diff
@@ -73,14 +88,21 @@ export const TableRow = function TableRow<T>({ row, group, ...props }: TableRowP
             : value;
         }
 
-        let classNames: string[] = [
-          col.className ?? '',
-          col.fixed ? `fixed fixed-${col.fixed}` : '',
-        ].filter(s => !!s);
+        let classNames: string[] = [col.className ?? '', col.fixed ? `fixed fixed-${col.fixed}` : ''];
+        let cellProps: DetailedHTMLProps<TdHTMLAttributes<HTMLTableCellElement>, HTMLTableCellElement> = {};
+        
+        if (tableCellProps && typeof tableCellProps === 'function'){
+          cellProps = tableCellProps(col, row) ?? {}
+          if(cellProps.key) delete cellProps.key;
 
-        return <td key={colIdx} className={classNames.join(' ').trim()}>
-          {rendered}
-        </td>;
+          if (cellProps.className) {
+            classNames.push(cellProps.className)
+          }
+          classNames.filter(s => !!s);
```

Comment:
`.filter` does not mutate the array - it would be more like:
```suggestion
          classNames = classNames.filter(s => !!s);
```